test/nodes/test_preprocessor.py

```diff
@@ -178,6 +219,75 @@ def test_preprocess_word_split():
     assert len(documents) == 15
 
 
+@pytest.mark.unit
+def test_preprocess_tiktoken_token_split(mock_tiktoken_tokenizer):
+    raw_docs = [
+        "This is a document. It has two sentences and eleven words.",
+        "This is a document with a long sentence (longer than my split length), it has seventeen words.",
+    ]
+    docs = [Document(content=content) for content in raw_docs]
+    split_length = 10
+    token_split_docs_not_respecting_sentences = PreProcessor(
+        split_by="token",
+        split_length=split_length,
+        split_respect_sentence_boundary=False,
+        split_overlap=0,
+        tokenizer="tiktoken",
+    ).process(docs)
+    assert len(token_split_docs_not_respecting_sentences) == 4
+    enc = tiktoken.get_encoding("cl100k_base")
+    split_documents_encoded = [
+        enc.encode(d.content, allowed_special="all", disallowed_special=())
+        for d in token_split_docs_not_respecting_sentences
+    ]
+    assert all([len(d) <= split_length for d in split_documents_encoded])
+    token_split_docs_respecting_sentences = PreProcessor(
+        split_by="token",
+        split_length=split_length,
+        split_respect_sentence_boundary=True,
+        split_overlap=0,
+        tokenizer="tiktoken",
+    ).process(docs)
+    assert len(token_split_docs_respecting_sentences) == 3  # should not be more than there are sentences
```

Comment:
Let's check the splits here too:

```suggestion
    assert token_split_docs_not_respecting_sentences == [
        "This is a document. ",
        "It has two sentences and eleven words.",
        "This is a document with a long sentence (longer than my split length), it has seventeen words."
    ]
```